contrib_bots/bots/generic/generic.py

```diff
@@ -0,0 +1,198 @@
+class BotCommandRegistry(object):
+    def __init__(self, about_text, just_mentioned=-1):
+        # Note the values are empty, since they are handled elsewhere
+        self._commands = \
+            {'help'    : ('','output this help text'),
+             'commands': ('','briefly list these supported commands'),
+             'about'   : ('','tells you what this bot is intended to do'),
+            }
+        self._integer_command = None
+        self._about_text = about_text
+        self._just_mentioned = just_mentioned
+      
+    def usage(self):
+        return '''
+            This generic zulip bot is intended to be a simple extensible design
+            which relies upon users simply adding in extra elements into a dict.
+            '''
+
+    def add_command(self, cmd, text_or_function, explanation):
+        if cmd not in self._commands.keys():
+            self._commands[cmd] = (text_or_function, explanation)
+        else:
+            print("WARNING: %s command already exists, so haven't added it." % cmd)
+
+    def add_int_handler(self, text_or_function, explanation):
+        self._integer_command = (text_or_function, explanation)
+        self._commands["<id>"] = ('', explanation)
+
+    def handle_message(self, message, client, state_handler):
+        msg = self._about_text
+
+        content = message['content']
+        if len(content) == 0:
+            if self._just_mentioned != -1:
+                if not isinstance(self._just_mentioned, str):
+                    msg = self._just_mentioned()
+                else:
+                    msg = self._just_mentioned
+                content = "about" # Use pass-through below
+            else:
+                content = "help" # Default handling of just mentioning bot
+            
+        command = content.split()[0].strip()
+        args = content.split()[1:]
+
+        if self._integer_command is not None and command.isdigit():
+            msg = self._integer_command[0](command,args)
+        elif command == 'about':
+            pass # defaults to about text
+        elif command == 'help':
+            for key, value in self._commands.items(): msg += "\n* " + key + " : " + value[1]
+        elif command == 'commands':
+            msg = "Commands: " + (" ".join(self._commands.keys()))
+        elif command == '<id>' and self._integer_command is not None:
+            msg = "This command is not intended to be used explicitly, but is a placeholder for a number; please check the help."
+        elif command in self._commands.keys():
+            if not isinstance(self._commands[command][0], str): 
+                msg = self._commands[command][0](args)
+            else:
+                msg = self._commands[command][0]
+        else:
+            msg = "Sorry, I don't understand that command. Please try one of the following.\n"
+            msg += "Commands: " + (" ".join(self._commands.keys()))
+
+        client.send_reply(message, msg)
+
+
+import time
```

Comment:
Yeah, this is specifically a WIP test to explore the idea of the framework, and since I started putting the code in a bot location I was testing it by putting the 'demo' code below it. If it becomes a framework then the two parts will be separated, so this import will be at the top of a file :)